internal/aws/client.go

```diff
@@ -54,6 +54,7 @@ type Client interface {
 	FindGroupByDisplayName(string) (*Group, error)
 	FindUserByEmail(string) (*User, error)
 	IsUserInGroup(*User, *Group) (bool, error)
+	UpdateUser(*User) (*User, error)
```

Comment:
I would suggest to not use references here, but return the values.

Reply:
I copied the logic from the other functions.  I kept the pointer in the function declaration but changed all functions to return values instead of named return values and referencs